show/main.py

```diff
@@ -1133,6 +1120,215 @@ def ntp(verbose):
+                snmp_user_auth_password = 'Null'
+            try:
+                snmp_user_encryption_type = snmp_users_keys[line]['SNMP_USER_ENCRYPTION_TYPE']
+            except KeyError:
+                snmp_user_encryption_type = 'Null'
+            try:
+                snmp_user_encryption_password = snmp_users_keys[line]['SNMP_USER_ENCRYPTION_PASSWORD']
+            except KeyError:
+                snmp_user_encryption_password = 'Null'
+            try:
+                snmp_user_type = snmp_users_keys[line]['SNMP_USER_TYPE']
+            except KeyError:
+                snmp_user_type = 'Null'
+            snmp_user_body.append([snmp_user, snmp_user_permissions_type, snmp_user_type, snmp_user_auth_type, 
+                                   snmp_user_auth_password, snmp_user_encryption_type, snmp_user_encryption_password])
+        click.echo(tabulate(natsorted(snmp_user_body), snmp_user_header))
+
+
+# ("show runningconfiguration snmp")
+@clicommon.pass_db
+def show_run_snmp_global(db, ctx):
+    snmp_global_table = db.cfgdb.get_table('SNMP')
+    snmp_comm_table = db.cfgdb.get_table('SNMP_COMMUNITY')
+    snmp_users_table = db.cfgdb.get_table('SNMP_USER')
+    snmp_location_header = ["Location"]
+    snmp_location_body = []
+    snmp_contact_header = ["SNMP_CONTACT", "SNMP_CONTACT_EMAIL"]
+    snmp_contact_body = []
+    snmp_comm_header = ["Community String", "Community Type"]
+    snmp_comm_body = []
+    snmp_user_header = ['User', "Permission Type", "Type", "Auth Type", "Auth Password", "Encryption Type", 
+                        "Encryption Password"]
+    snmp_user_body = []
+    try:
+        if snmp_global_table['LOCATION']:
+            snmp_location = [snmp_global_table['LOCATION']['Location']]
+            snmp_location_body.append(snmp_location)
+    except KeyError:
+        snmp_global_table['LOCATION'] = ''
+    click.echo(tabulate(snmp_location_body, snmp_location_header))
+    click.echo("\n")
+    try:
+        if snmp_global_table['CONTACT']:
+            snmp_contact = list(snmp_global_table['CONTACT'].keys())
+            snmp_contact_email = [snmp_global_table['CONTACT'][snmp_contact[0]]]
+            snmp_contact_body.append([snmp_contact[0], snmp_contact_email[0]])
+    except KeyError:
+        snmp_global_table['CONTACT'] = ''
+    click.echo(tabulate(snmp_contact_body, snmp_contact_header))
+    click.echo("\n")
+    snmp_comm_strings = snmp_comm_table.keys()
+    snmp_users = snmp_users_table.keys()
+    for line in snmp_comm_strings:
+        comm_string = line
+        comm_string_type = snmp_comm_table[line]['TYPE']
+        snmp_comm_body.append([comm_string, comm_string_type])
+    click.echo(tabulate(natsorted(snmp_comm_body), snmp_comm_header))
+    click.echo("\n")
+    for line in snmp_users:
+        snmp_user = line
+        try:
+            snmp_user_permissions_type = snmp_users_table[line]['SNMP_USER_PERMISSION']
+        except KeyError:
+            snmp_user_permissions_type = 'Null'
```

Comment:
>'Null' [](http://example.com/codeflow?start=41&length=6)

Use a named constant?

Reply:
Are you saying you'd like me to have something like this? 
UNSET_USER_VAR = "Null"? 
and then in each except KeyError set the snmp_user parameter to that instead?

Reply:
Changed the way I was doing this per what you had above.  I don't think this is needed anymore.